opentelemetry-sdk/src/metrics/instrument.rs

```diff
@@ -253,25 +253,25 @@ pub(crate) struct ResolvedMeasures<T> {
 }
 
 impl<T: Copy + 'static> SyncCounter<T> for ResolvedMeasures<T> {
-    fn add(&self, val: T, attrs: &[KeyValue]) {
+    fn add(&self, val: T, attrs: AttributeSet) {
```

Comment:
+1 on making adding a generic function here to replace `AttributeSet`

Reply:
Made the outer `Counter` accept the `Into<AttributeSet>` but left the inner `Sync*` types alone.